test/asynchronous/test_retryable_reads.py

```diff
@@ -0,0 +1,196 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test retryable reads spec."""
+from __future__ import annotations
+
+import os
+import pprint
+import sys
+import threading
+
+from pymongo.errors import AutoReconnect
+
+sys.path[0:0] = [""]
+
+from test.asynchronous import (
+    AsyncIntegrationTest,
+    AsyncPyMongoTestCase,
+    async_client_context,
+    client_knobs,
+    unittest,
+)
+from test.utils import (
+    CMAPListener,
+    OvertCommandListener,
+    async_set_fail_point,
+)
+
+from pymongo.monitoring import (
+    ConnectionCheckedOutEvent,
+    ConnectionCheckOutFailedEvent,
+    ConnectionCheckOutFailedReason,
+    PoolClearedEvent,
+)
+
+_IS_SYNC = False
+
+# Location of JSON test specifications.
+_TEST_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), "retryable_reads", "legacy")
+
+
+class TestClientOptions(AsyncPyMongoTestCase):
+    async def test_default(self):
+        client = self.simple_client(connect=False)
+        self.assertEqual(client.options.retry_reads, True)
+
+    async def test_kwargs(self):
+        client = self.simple_client(retryReads=True, connect=False)
+        self.assertEqual(client.options.retry_reads, True)
+        client = self.simple_client(retryReads=False, connect=False)
+        self.assertEqual(client.options.retry_reads, False)
+
+    async def test_uri(self):
+        client = self.simple_client("mongodb://h/?retryReads=true", connect=False)
+        self.assertEqual(client.options.retry_reads, True)
+        client = self.simple_client("mongodb://h/?retryReads=false", connect=False)
+        self.assertEqual(client.options.retry_reads, False)
+
+
+class FindThread(threading.Thread):
+    def __init__(self, collection):
+        super().__init__()
+        self.daemon = True
+        self.collection = collection
+        self.passed = False
+
+    async def run(self):
+        await self.collection.find_one({})
+        self.passed = True
+
+
+class TestPoolPausedError(AsyncIntegrationTest):
+    # Pools don't get paused in load balanced mode.
+    RUN_ON_LOAD_BALANCER = False
+    RUN_ON_SERVERLESS = False
+
+    @async_client_context.require_sync
+    @async_client_context.require_failCommand_blockConnection
+    @client_knobs(heartbeat_frequency=0.05, min_heartbeat_interval=0.05)
+    async def test_pool_paused_error_is_retryable(self):
+        if "PyPy" in sys.version:
+            # Tracked in PYTHON-3519
+            self.skipTest("Test is flakey on PyPy")
+        cmap_listener = CMAPListener()
+        cmd_listener = OvertCommandListener()
+        client = await self.async_rs_or_single_client(
+            maxPoolSize=1, event_listeners=[cmap_listener, cmd_listener]
+        )
+        self.addAsyncCleanup(client.close)
```

Comment:
We don't need to add explicit cleanup when using the helper methods to create clients, they do the cleanup for us.